apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/hooks/useEstimateMarketWithdraw.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { parseUsdPrice, parseNumberUsdPriceToBigInt } from '@common/utils/currency';
+import { formatUnits, parseUnits } from 'viem';
+import { calculateEarnFeeAmount } from '@common/utils/earn/parsing';
+import useEarnService from '@hooks/earn/useEarnService';
+import { useEarnManagementState } from '@state/earn-management/hooks';
+import { AmountsOfToken, DisplayStrategy, EarnPosition, FeeType } from 'common-types';
+
+interface UseEstimateMarketWithdrawProps {
+  strategy?: DisplayStrategy;
+  shouldRefetch: boolean;
+  position?: EarnPosition;
+}
+
+export default function useEstimateMarketWithdraw({
+  strategy,
+  shouldRefetch,
+  position,
+}: UseEstimateMarketWithdrawProps) {
+  const earnService = useEarnService();
+  const [estimatedMarketAmount, setEstimatedMarketAmount] = React.useState<AmountsOfToken | undefined>();
+  const [isLoading, setIsLoading] = React.useState(false);
```

Comment:
shouldn't we be using the lastest pattern of having { result, isLoading, error} here? It triggers less rerenders